### PR TITLE
Feature/make integration tests independent

### PR DESCRIPTION
**Pull request by vxsx**
- Added `cms` helpers to quickly login/logout/add/remove page/addplugin
- added setup / teardown for every test, so they can be run separately
- changed some tests because they were a bit too strict

### CODE REVIEW

File: cms/tests/frontend/integration/addFirstPage.js [hunk 1]
```diff
@@ -6,6 +6,18 @@
...
 var messages = require('./settings/messages').page.creation;
+var cms = require('./helpers/cms')();
+
+casper.test.setUp(function (done) {
+    casper.start().then(cms.login).run(done);
+});
+
+casper.test.tearDown(function (done) {
+    casper.start()
+        .then(cms.removeFirstPage)
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/addFirstPage.js]: don't we want a method like: cms.removePage({ first: true }); or smth?

File: cms/tests/frontend/integration/createContent.js [hunk 2]
```diff
@@ -51,18 +68,24 @@ casper.test.begin('User Add Content', function (test) {
...
+                        return $('.cms-submenu-item [data-rel="add"]:visible').length === 1;
+                    });
+                }).then(function () {
+                    test.assertVisible('.cms-plugin-picker .cms-submenu-item [data-rel="add"]',
+                        messages.filteredPluginAvailable);
+                });
             });
             this.then(function () {
-                this.click('.cms-plugin-picker .cms-submenu-item [data-rel="add"]');
+                this.click(xPath('//a[@data-rel="add"]/text()[normalize-space(.)="Text"]'));
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/createContent.js]: what does this do?

File: cms/tests/frontend/integration/helpers/cms.js [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+            return this.thenOpen(globals.adminUrl).then(function () {
+                this.fill('#login-form', globals.credentials, true);
+            });
+        },
+
+        logout: function () {
+            return this.thenOpen(globals.adminLogoutUrl);
+        },
+
+        removeFirstPage: function () {
```
Comment (FinalAngel) [inline cms/tests/frontend/integration/helpers/cms.js]: I would prefer a general "removePage" method

File: cms/tests/frontend/integration/addFirstPage.js (hunk 1 above)
Comment (vxsx) [inline cms/tests/frontend/integration/addFirstPage.js]: yeah in general we would want to do a generic removePage, to be able to remove by name or slug
makes sense

File: cms/tests/frontend/integration/createContent.js (hunk 2 above)
Comment (vxsx) [inline cms/tests/frontend/integration/createContent.js]: XPath is <3
it finds all links with attribute `data-rel="add"` which contain text that after trimming spaces from both sides is equal to `Text`

**Comment by dhorelik** [comment]
This. Is. Awesome. :+1:

File: cms/tests/frontend/integration/createContent.js (hunk 2 above)
Comment (FinalAngel) [inline cms/tests/frontend/integration/createContent.js]: :+1:

**Comment by lory87** [comment]
:+1:

**Comment by FinalAngel** [comment]
LGTM :+1:

**Pull request merged**